### PR TITLE
ACP-2626 Stripe marketplace payment flow marketplace oms workflow transfers to merchants

### PR DESCRIPTION
**Pull request by stereomon**
## PR Description

TBD

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -63,6 +63,12 @@
             "Orm\\": "src/Orm/"
         }
     },
+    "repositories": [
+        {
+            "type": "git",
+            "url": "git@github.com:spryker/app-payment.git"
+        }
+    ],
```
Comment (matweew) [inline composer.json]: not needed already, I see it in Packagist https://packagist.org/packages/spryker/app-payment

File: src/Spryker/Zed/AppMerchant/Persistence/AppMerchantRepository.php [hunk 2]
```diff
@@ -32,4 +32,27 @@ public function findMerchant(MerchantCriteriaTransfer $merchantCriteriaTransfer)
         return $this->getFactory()->createMerchantMapper()
             ->mapMerchantEntityToMerchantTransfer($merchantEntity, new MerchantTransfer());
     }
+
+    /**
+     * @return array<\Generated\Shared\Transfer\MerchantTransfer>
+     */
+    public function findMerchants(MerchantCriteriaTransfer $merchantCriteriaTransfer): array
```
Comment (matweew) [inline src/Spryker/Zed/AppMerchant/Persistence/AppMerchantRepository.php:39]: Note: `findMerchant()` is a special case of `findMerchants()`

File: src/Spryker/Zed/AppMerchant/AppMerchantConfig.php [hunk 3]
```diff
@@ -33,12 +33,24 @@ public function getAppIdentifier(): string
...
+
+    /**
+     * @api
+     *
+     * @return array<string>
+     */
+    public function getHandleableWebhookTypes(): array
+    {
+        return [
+            'account',
```
Comment (matweew) [inline src/Spryker/Zed/AppMerchant/AppMerchantConfig.php:55]: where is this type originally defined? maybe some constant in app-webhook?

File: src/Spryker/Zed/AppMerchant/Business/AppMerchantFacadeInterface.php [hunk 4]
```diff
@@ -57,6 +58,18 @@ public function createMerchantAppOnboarding(
      */
     public function handleWebhook(WebhookRequestTransfer $webhookRequestTransfer, WebhookResponseTransfer $webhookResponseTransfer): WebhookResponseTransfer;
 
+    /**
+     * Specification:
+     * - Filters all order items that do not have a merchantReference.
+     * - Groups order items by order reference and merchant reference.
+     * - Creates a new `PaymentsTransmissionsRequestTransfer` with the order items grouped by order reference and merchant reference.
```
Comment (matweew) [inline src/Spryker/Zed/AppMerchant/Business/AppMerchantFacadeInterface.php]:
```suggestion
     * - Extends  `PaymentsTransmissionsRequestTransfer` with the order items grouped by order reference and merchant reference.
```

File: src/Spryker/Zed/AppMerchant/Persistence/Propel/Schema/spy_app_payment.schema.xml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
+<database xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="zed" xsi:noNamespaceSchemaLocation="http://static.spryker.com/schema-01.xsd" namespace="Orm\Zed\AppPayment\Persistence" package="src.Orm.Zed.AppPayment.Persistence">
+
+    <table name="spy_payment_transfer" idMethod="native" class="SpyPayment">
+        <column name="merchant_reference" type="CHAR" size="36" required="false" description="Has to be false for order items that do not have a merchant"/>
```
Comment (matweew) [inline src/Spryker/Zed/AppMerchant/Persistence/Propel/Schema/spy_app_payment.schema.xml]: >Has to be false for order items that do not have a merchant
"false" for the CHAR field? maybe empty/null?

File: src/Spryker/Zed/AppMerchant/AppMerchantConfig.php (hunk 3 above)
Comment (stereomon) [reply src/Spryker/Zed/AppMerchant/AppMerchantConfig.php:55]: Constants are in the AppPayment. The real values will be defined in the App implementation. This needs documentation.

File: src/Spryker/Zed/AppMerchant/Persistence/Propel/Schema/spy_app_payment.schema.xml (hunk 5 above)
Comment (stereomon) [reply src/Spryker/Zed/AppMerchant/Persistence/Propel/Schema/spy_app_payment.schema.xml]: The text is misleading, true. Updated

**Review by matweew — APPROVED** [review]

**Pull request merged**